### PR TITLE
Webpack migration and updates

### PR DESCRIPTION
**Pull request by blackleaden**
Current status: packages updated, code updated, incl. corejs and ts. 
JQuery removed. 
Debug in process.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,36 +10,41 @@
     "url": "git+https://github.com/Microsoft/powerbi-visuals-sampleslicer"
   },
   "scripts": {
-    "postinstall": "pbiviz update 1.13.0",
+    "postinstall": "pbiviz update",
```
Comment (uve) [inline package.json]: remove this line

File: pbiviz.json [hunk 2]
```diff
@@ -2,35 +2,20 @@
...
+    "guid": "SampleSlicer1448559807355TEST5",
     "visualClassName": "SampleSlicer",
-    "version": "1.3.0",
+    "version": "2.0.5",
     "description": "Use this slicer to select discrete numeric categories or range of numeric category values.",
     "supportUrl": "http://community.powerbi.com",
     "gitHubUrl": "https://github.com/Microsoft/powerbi-visuals-sampleslicer"
   },
-  "apiVersion": "1.13.0",
+  "apiVersion": "2.5.0",
```
Comment (uve) [inline pbiviz.json]: set API v2.6.0

File: src/sampleSlicer.ts [hunk 3]
```diff
@@ -24,840 +24,960 @@
...
-    import textMeasurementService = powerbi.extensibility.utils.formatting.textMeasurementService;
-
-    import FilterManager = powerbi.extensibility.utils.filter.FilterManager;
-    import AppliedFilter = powerbi.extensibility.utils.filter.AppliedFilter;
-
-    export const enum RangeValueType {
-        Start,
-        End
-    }
+// "use strict";
```
Comment (uve) [inline src/sampleSlicer.ts]: remove comments

File: src/sampleSlicer.ts [hunk 4]
```diff
@@ -24,840 +24,960 @@
...
-                !dataView2.categorical) {
-                return false;
-            }
+        return true;
+    }
 
-            let dv1Categories: DataViewCategoricalColumn[] = dataView1.categorical.categories;
-            let dv2Categories: DataViewCategoricalColumn[] = dataView2.categorical.categories;
+    /*
+     *  Public
```
Comment (uve) [inline src/sampleSlicer.ts]: remove comments

File: src/sampleSlicer.ts [hunk 5]
```diff
@@ -24,840 +24,960 @@
...
+     */
 
-            if (!dv1Categories ||
-                !dv2Categories ||
-                dv1Categories.length !== dv2Categories.length) {
-                return false;
-            }
+    constructor(options: VisualConstructorOptions) {
+        if (window.location !== window.parent.location) {
```
Comment (uve) [inline src/sampleSlicer.ts]: remove core-js usage

File: src/sampleSlicer.ts [hunk 6]
```diff
@@ -24,840 +24,960 @@
...
-                this.behavior.scalableRange,
-                this.visualHost);
+        this.updateInternal(categoryIdentityChanged);
+        this.eventService.renderingFinished(options);
+    }
 
-            if (!data) {
-                this.tableView.empty();
+    /*
+     *  Private
```
Comment (uve) [inline src/sampleSlicer.ts]: remove comments

File: src/sampleSlicer.ts [hunk 7]
```diff
@@ -24,840 +24,960 @@
...
-                data.slicerDataPoints.filter(x => !x.filtered),
-                (d: SampleSlicerDataPoint) => $.inArray(d, data.slicerDataPoints),
-                resetScrollbarPosition)
-                .viewport(this.getSlicerBodyViewport(this.currentViewport))
-                .render();
-
-            this.updateSliderControl();
-            this.updateSliderInputTextboxes();
-        }
+        // this.initClearButton(outerContainer); //Temporary unavailable
```
Comment (uve) [inline src/sampleSlicer.ts]: remove comments

File: src/selectionBehavior.ts [hunk 8]
```diff
@@ -24,133 +24,188 @@
...
+    private slicerSettings: Settings;
+    private options: SampleSlicerBehaviorOptions;
+    private dataPoints: SampleSlicerDataPoint[];
+    private callbacks: SampleSlicerCallbacks;
+
+    private static mapDataPointsToFilterValues (dataPoints: SampleSlicerDataPoint[]): any[] {
+        return (dataPoints
+            .map( (dataPoint: SampleSlicerDataPoint) => (dataPoint.category || null) ) || [])
+            .filter( (value: string | number | null ) => (typeof value === 'string' || typeof value === 'number') )
+            //.map( (value: string | number  | null) => ((!isNaN(Number(value)) ? Number(value) : value ) ) );
```
Comment (uve) [inline src/selectionBehavior.ts]: remove comments

File: src/selectionBehavior.ts [hunk 9]
```diff
@@ -24,133 +24,188 @@
...
+    public bindEvents(options: SampleSlicerBehaviorOptions, selectionHandler: ISelectionHandler): void {
+        const slicers: Selection<FilterDataPoint> = this.slicers = options.slicerItemContainers;
+
+        this.dataPoints = options.dataPoints;
+        this.interactivityService = options.interactivityService;
+        this.slicerSettings = options.slicerSettings;
+        this.options = options;
+
+        this.selectionHandler = selectionHandler;
+        // (this: T, datum: Datum, index: number, groups: T[] | ArrayLike<T>) => Result
```
Comment (uve) [inline src/selectionBehavior.ts]: remove comments

File: src/tableView.ts [hunk 10]
```diff
@@ -24,339 +24,342 @@
...
+    public render(): void {
+        let options: TableViewOptions = this.options,
+            visibleGroupContainer: Selection<any> = this.visibleGroupContainer,
+            rowHeight: number = options.rowHeight || TableView.defaultRowHeight,
+            groupedData: TableViewGroupedData = this.getGroupedData(),
+            rowSelection: Selection<any>,
+            cellSelection: Selection<any>;
+        
+        let i = 0;
+        for (i = 0;  i<6; i++) { // UPDATE FIX
```
Comment (uve) [inline src/tableView.ts]: what it's magic 6 number? please fix it.

File: pbiviz.json [hunk 11]
```diff
@@ -2,35 +2,20 @@
   "visual": {
     "name": "sampleSlicer",
     "displayName": "SampleSlicer",
-    "guid": "SampleSlicer1448559807355",
+    "guid": "SampleSlicer1448559807355TEST5",
```
[the remark this answers is not in the document]
Comment (blackleaden) [reply pbiviz.json]: done

File: src/sampleSlicer.ts (hunk 3 above)
Comment (blackleaden) [reply src/sampleSlicer.ts]: done

File: src/sampleSlicer.ts (hunk 4 above)
Comment (blackleaden) [reply src/sampleSlicer.ts]: done

File: src/sampleSlicer.ts (hunk 5 above)
Comment (blackleaden) [reply src/sampleSlicer.ts]: done

File: src/sampleSlicer.ts (hunk 6 above)
Comment (blackleaden) [reply src/sampleSlicer.ts]: done

File: src/tableView.ts (hunk 10 above)
Comment (blackleaden) [reply src/tableView.ts]: fixed

**Pull request merged**